### PR TITLE
harden cluster leaving, #21847

### PR DESCRIPTION
**Pull request by patriknw**
As documented in the code:

// Leader is moving itself from Leaving to Exiting. Let others know (best effort)
// before shutdown. Otherwise they will not see the Exiting state change
// and there will not be convergence until they have detected this node as
// unreachable and the required downing has finished. They will still need to detect
// unreachable, but Exiting unreachable will be removed without downing, i.e.
// normally the leaving of a leader will be graceful without the need
// for downing. However, if those final gossip messages never arrive it is
// alright to require the downing, because that is probably caused by a
// network failure anyway.

That is fine, but this change improves the selection of the nodes to
send the final gossip messages to.

I could reproduce the failure in ClusterSingletonManagerLeaveSpec and with
additional logging I verified that in the failure case it picked the "first"
node 3 times (it's random) and that node had already been shutdown (left earlier
in the test) but was not removed yet.

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #21847

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 1]
```diff
@@ -776,6 +780,32 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef) extends Actor with
     (latestGossip.overview.seen.size < latestGossip.members.size / 2)
 
   /**
+   * Sends full gossip to `n` other random members.
+   */
+  def gossipRandomN(n: Int): Unit = {
+    if (!isSingletonCluster) {
```
Comment (viktorklang) [inline akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala]: `&& n > 0`

**Review by johanandren — APPROVED** [review]

**Comment by drewhk** [comment]
LGTM

**Comment by ktoso** [comment]
LGTM, applying Viktor's suggestion would be nice too

**Comment by patriknw** [comment]
I included n > 0

**Pull request merged**